### PR TITLE
MAT-1484 add constructors to initialize resource type

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: test/integration/JsonSerialization.test.ts [hunk 1]
```diff
@@ -10,6 +10,7 @@
   PrimitiveUri,
   ResourceChild,
 } from "../fixture/generatedTypeScript/fhir";
+import { ValueSet } from "../../generated/typescript/classes/ValueSet";
```
Comment (AndrewBird81) [inline test/integration/JsonSerialization.test.ts:13]: Codacy found an issue: [Missing file extension for "../../generated/typescript/classes/ValueSet"](https://app.codacy.com/gh/MeasureAuthoringTool/model-info-parser/pullRequest?prid=6104340)

**Comment by AndrewBird81** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 6
           
```


See the complete overview on [Codacy](https://app.codacy.com/gh/MeasureAuthoringTool/model-info-parser/pullRequest?prid=6104340&bid=19982264)

File: test/integration/JsonSerialization.test.ts [hunk 2]
```diff
@@ -147,4 +148,10 @@
   }
 }`);
   });
+
+  it("should initialize resource type by default", () => {
+    const valueSet: ValueSet = new ValueSet();
+    expect(valueSet.resourceType).toEqual("ValueSet");
```
Comment (AndrewBird81) [inline test/integration/JsonSerialization.test.ts:154]: Codacy found an issue: [Unsafe member access .toEqual on an any value.](https://app.codacy.com/gh/MeasureAuthoringTool/model-info-parser/pullRequest?prid=6104340)

File: test/integration/JsonSerialization.test.ts [hunk 3]
```diff
@@ -147,4 +148,10 @@
   }
 }`);
   });
+
+  it("should initialize resource type by default", () => {
+    const valueSet: ValueSet = new ValueSet();
```
Comment (AndrewBird81) [inline test/integration/JsonSerialization.test.ts:153]: Codacy found an issue: [Unsafe construction of an any type value.](https://app.codacy.com/gh/MeasureAuthoringTool/model-info-parser/pullRequest?prid=6104340)

File: test/integration/JsonSerialization.test.ts (hunk 2 above)
Comment (AndrewBird81) [inline test/integration/JsonSerialization.test.ts:154]: Codacy found an issue: [Unsafe call of an any typed value.](https://app.codacy.com/gh/MeasureAuthoringTool/model-info-parser/pullRequest?prid=6104340)

File: test/integration/JsonSerialization.test.ts (hunk 3 above)
Comment (AndrewBird81) [inline test/integration/JsonSerialization.test.ts:153]: Codacy found an issue: [Unsafe assignment of an any value.](https://app.codacy.com/gh/MeasureAuthoringTool/model-info-parser/pullRequest?prid=6104340)

File: test/integration/JsonSerialization.test.ts [hunk 4]
```diff
@@ -147,4 +148,10 @@
   }
 }`);
   });
+
+  it("should initialize resource type by default", () => {
```
Comment (AndrewBird81) [inline test/integration/JsonSerialization.test.ts:152]: Codacy found an issue: [Unsafe call of an any typed value.](https://app.codacy.com/gh/MeasureAuthoringTool/model-info-parser/pullRequest?prid=6104340)

**Review by AndrewBird81 — CHANGES_REQUESTED** [review]
I like this approach! It's a lot simpler than what I was envisioning. But you shouldn't rely on the "generated" classes, as those aren't checked in. You need to re-generate the test fixtures. You can simply run `npm run-script build`

**Comment by serhii-ilin** [comment]
> I like this approach! It's a lot simpler than what I was envisioning. But you shouldn't rely on the "generated" classes, as those aren't checked in. You need to re-generate the test fixtures. You can simply run `npm run-script build`

I ran `npm run-script build` and pushed the changes.

**Review by adongare — DISMISSED** [review]

**Comment by AndrewBird81** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 6
           
```


See the complete overview on [Codacy](https://app.codacy.com/gh/MeasureAuthoringTool/model-info-parser/pullRequest?prid=6104340&bid=19982264)

**Review by AndrewBird81 — APPROVED** [review]

**Review by adongare — APPROVED** [review]

**Pull request merged**